From: Josh Law <hlcj1234567@gmail.com>
Subject: lib: glob: replace bitwise OR with logical operation on boolean
Date: Sun, 1 Mar 2026 15:21:42 +0000

Using bitwise OR (|=) on a boolean variable is valid C, but replacing it
with a direct logical assignment makes the intent clearer and appeases
strict static analysis tools.

Link: https://lkml.kernel.org/r/20260301152143.2572137-2-objecting@objecting.org
Signed-off-by: Josh Law <objecting@objecting.org>
Reviewed-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/glob.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/lib/glob.c~lib-glob-replace-bitwise-or-with-logical-operation-on-boolean
+++ a/lib/glob.c
@@ -96,7 +96,8 @@ bool __pure glob_match(char const *pat,
 					class += 2;
 					/* Any special action if a > b? */
 				}
-				match |= (a <= c && c <= b);
+				if (a <= c && c <= b)
+					match = true;
 			} while ((a = *class++) != ']');
 
 			if (match == inverted)
_
